src/sci/impl/analyzer.cljc

```diff
@@ -495,7 +495,9 @@
                  iden->invoke-idx (:iden->invoke-idx ctx)
                  iden->invoke-idx (assoc iden->invoke-idx new-iden idx)
                  ctx (assoc ctx :iden->invoke-idx iden->invoke-idx)]
-             [(update ctx :bindings assoc binding-name new-iden)
+             [(update ctx :bindings #(-> %
```

Comment:
If this isn't causing a major perf hit, I'm fine with accepting this. May want to benchmark on some typical examples.

Reply:
Yeah, a good idea to compare before and after.

Do you have a typical methodology when doing that?

Reply:
Not really, but you could e.g. try `(sci/eval-form '(fn [] (let [x 1 y 2 z 3] nil)))` or so and do this in a dotimes. Better would be to isolate the analysis. I don't think in the grand scheme that this will matter a lot. 

Btw, does the error also occur with nested fns, and should we thus also apply this fix in fn locals?

Reply:
Did not test nested fns.

Nor deeply nested `let`s.

What about `(def x)` vs `(defn some-fun [x])`? Is that something to also look at?

Reply:
Although I'm not certain of the implications. You may want to make two different PRs (even incomplete). I'll get to this myself after my vacation (don't have enough time in the day to go deep into this). Perhaps the conclusion will be that the `dissoc`-first approach is still best.

Reply:
Yeah, I'll add more tests to cover:
1. nested fns
2. `(def ^String x "boo") (defn some-fun [^Integer x])`
3. ?

... and adapt the current approach as necessary.

I'll also try a separate PR for new-iden approach.  
I'm assuming new-iden will be something that can always hold metadata, but that will become clear to me as I explore.

Reply:
Yes, new-iden is also a symbol, which is basically a renamed binding name to make it unique